FrontEnd/src/UserAuthentication/SignUpPage.jsx

```
@@ -1,6 +1,30 @@
 import  { useNavigate } from "react-router-dom"
+import {useState} from "react";
+import {registerUser} from "./Auth";
+
 function SignUpPage(){
     let navigate = useNavigate();
+    const [signUpEmail, setSignUpEmail] = useState('');
+    const [signUpPassword, setSignUpPassword] = useState('');
+    const minimumPasswordLength = 6;
+
+    const handleSignUp = async(event) =>{
+        event.preventDefault();
+        if (signUpPassword.length < minimumPasswordLength) {
+            alert(`Minimum password length is ${minimumPasswordLength}`)
+        } else {
+            try{
+                const result = await registerUser(signUpEmail, signUpPassword);
+                if (result !== null){
+                    navigate('/MainPage')
+                } else {
+                    console.error('Error creating user');
```

Let's throw a new exception here, so we catch the error in one error console.log.